src/http-blockstore.ts

```diff
@@ -0,0 +1,109 @@
+import type { Blockstore, Pair } from 'interface-blockstore'
+import type { Connection } from '@libp2p/interface-connection'
+import type {
+  AbortOptions,
+  Await,
+  AwaitIterable
+} from 'interface-store'
+import type { Libp2p } from 'libp2p'
+import type { CID } from 'multiformats/cid'
+import { fetchViaDuplex } from '@marcopolo_/libp2p-fetch'
+
+// TODOS:
+// - Validate the blocks we fetch
+// - Use format=car to get the whole dag at once rather than per block (probably faster?)
+// - Keep a list of HTTPS endpoints, not just libp2p peers.
```

Comment:
I'd say we should (if we have time) just make up some record data and chuck it into IPNI for our records where the gateways are non-recursive (i.e. only based on data we have locally similar to https://github.com/ipfs/kubo/blob/c58aadb887a5ca88256ac8f3a8fbf60b0263c978/docs/config.md#gatewaynofetch).

Happy to help here, it shouldn't be too bad as I've done some hacking here before. Basically we'd use this format https://github.com/ipni/specs/pull/6 to avoid dealing with the global code table (and associated bikeshedding) as much as possible. It's a small amount of parsing in the code here, and then we can hook up some hacky IPNI record publishing.

For recursive gateways (e.g. dweb.link, ipfs.io, cf-ipfs.com) we could either hard code them, get them from https://ipfs.github.io/public-gateway-checker/ or have an IPNI rendezvous record.

----

If there's no time, then hard coding it is 😄